### PR TITLE
perf(sphereBrush): tool optimization

### PR DESCRIPTION
**Pull request by NeilMacPhee**
**Summary**
This fix optimizes the sphere tool for creating segmentations, reducing the time it takes for the texture to update. 

- "fillSphere.ts" now passes an array of modified slices to be rendered rather than rendering every slice in the volume
- "pointInSphere.ts" is now able to be optimized by the browser

### CODE REVIEW

File: packages/tools/src/tools/segmentation/strategies/fillSphere.ts [hunk 1]
```diff
@@ -44,7 +46,23 @@ function fillSphere(
...
+  const minSlice = Math.floor(scalarIndex[0] / zMultiple);
+  const maxSlice = Math.floor(scalarIndex[scalarIndex.length - 1] / zMultiple);
+  const testArray = Array.from(
+    { length: maxSlice - minSlice + 1 },
+    (v, k) => k + minSlice
+  );
+
+  const wrongSliceArray = Array.from({ length: 21 });
+
+  // console.log(testArray);
```
Comment (sedghi) [inline packages/tools/src/tools/segmentation/strategies/fillSphere.ts:59]: remove

File: packages/tools/src/tools/segmentation/strategies/fillSphere.ts [hunk 2]
```diff
@@ -44,7 +46,23 @@ function fillSphere(
     viewport as Types.IVolumeViewport
   );
 
-  triggerSegmentationDataModified(segmentationId);
+  const zMultiple = dimensions[0] * dimensions[1];
+  const minSlice = Math.floor(scalarIndex[0] / zMultiple);
+  const maxSlice = Math.floor(scalarIndex[scalarIndex.length - 1] / zMultiple);
+  const testArray = Array.from(
```
Comment (sedghi) [inline packages/tools/src/tools/segmentation/strategies/fillSphere.ts:52]: rename

File: packages/tools/src/tools/segmentation/strategies/fillSphere.ts [hunk 3]
```diff
@@ -44,7 +46,23 @@ function fillSphere(
...
+    { length: maxSlice - minSlice + 1 },
+    (v, k) => k + minSlice
+  );
+
+  const wrongSliceArray = Array.from({ length: 21 });
```
Comment (sedghi) [inline packages/tools/src/tools/segmentation/strategies/fillSphere.ts:57]: rename

File: packages/tools/src/tools/segmentation/strategies/fillSphere.ts [hunk 4]
```diff
@@ -44,7 +46,23 @@ function fillSphere(
...
+
+  // console.log(testArray);
+  const start = performance.now();
+
+  triggerSegmentationDataModified(segmentationId, testArray);
+
+  const end = performance.now();
+  console.log(`Execution time: ${end - start} ms`);
```
Comment (sedghi) [inline packages/tools/src/tools/segmentation/strategies/fillSphere.ts:65]: remove

**Review by sedghi — CHANGES_REQUESTED** [review]
This doesn't look like the correct PR, so many noise on the code + doesn't have the optimization of the fillSphere. Did you push the correct branch?

**Comment by NeilMacPhee** [comment]
> 

Apologizes, it looks like I forgot to push my latest changes before submitting the PR.

File: packages/tools/src/tools/segmentation/strategies/fillSphere.ts (hunk 1 above)
Comment (NeilMacPhee) [reply packages/tools/src/tools/segmentation/strategies/fillSphere.ts]: Removed wrongSliceArray

File: packages/tools/src/tools/segmentation/strategies/fillSphere.ts (hunk 2 above)
Comment (NeilMacPhee) [reply packages/tools/src/tools/segmentation/strategies/fillSphere.ts]: testArray has been renamed to sliceArray

File: packages/tools/src/tools/segmentation/strategies/fillSphere.ts (hunk 4 above)
Comment (NeilMacPhee) [reply packages/tools/src/tools/segmentation/strategies/fillSphere.ts]: performance markers have been removed

**Pull request merged**